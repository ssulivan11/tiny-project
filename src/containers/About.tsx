import { h, FunctionalComponent, Fragment } from "preact";

export interface Props {
  title: string;
}

export const About: FunctionalComponent<Props> = ({ title }: Props) => (
  <Fragment>
    <h1>{title}</h1>
    <p>
      This is just a simple{" "}
      <a href="https://github.com/preactjs/preact">preact</a> typescript demo
      project messing with{" "}
      <a href="https://github.com/vuejs/vite#bare-module-resolving">vite</a> for
      the dev server, stupid fast, no bundling and hmr.
    </p>
    <p>
      Also utilizing{" "}
      <a href="https://github.com/parcel-bundler/parcel">parcel</a> as the
      bundler. <a href="https://github.com/facebook/jest">Jest</a> and{" "}
      <a href="https://github.com/cypress-io/cypress">cypress</a> for testing.
    </p>
  </Fragment>
);
