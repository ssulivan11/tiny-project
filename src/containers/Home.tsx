import { Link } from "preact-router";
import { h, FunctionalComponent } from "preact";

export interface Props {
  title: string;
}

const imgStyle = { margin: "0 auto", display: "block" };

export const Home: FunctionalComponent<Props> = ({ title }: Props) => (
  <div className="text-center">
    <img
      style={imgStyle}
      src="src/assets/android-chrome-512x512.png"
      alt="Smiley face"
      height="200"
      width="200"
    />
    <h1>{title}</h1>
    <p>
      Read a little more <Link href="/about">here</Link>
    </p>
  </div>
);
