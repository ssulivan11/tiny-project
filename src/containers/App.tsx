import Router, { Link } from "preact-router";
import { h, FunctionalComponent } from "preact";
import { useState } from "preact/hooks";

// @ts-ignore
import { Home } from "./Home.tsx";
// @ts-ignore
import { About } from "./About.tsx";

export const App: FunctionalComponent = () => {
  const [active, setActive] = useState("home");

  return (
    <div>
      <nav class="nav">
        <div class="container">
          <Link
            onClick={() => setActive("home")}
            className={`${active === "home" ? "active" : ""} pagename current`}
            href="/"
          >
            TinyProject
          </Link>
          <Link
            onClick={() => setActive("about")}
            className={active === "about" ? "active" : ""}
            href="/about"
          >
            About
          </Link>
          <a href="https://github.com/ssulivan11/tiny-project">Github</a>
        </div>
      </nav>
      <div class="container">
        <Router>
          <Home path="/" title={"Welcome to Tiny Project!"} />
          <About path="/about" title={"About the Tiny Project"} />
        </Router>
      </div>
    </div>
  );
};
