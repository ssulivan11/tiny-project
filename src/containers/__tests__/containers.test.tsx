/// <reference types="enzyme-adapter-preact-pure" />

import { h } from "preact";
import { shallow, mount } from "enzyme";
import { App } from "../App";
import { Home } from "../Home";
import { About } from "../About";

test("Home tests", () => {
  const HomeContainer = shallow(<Home title="Testing" />);

  expect(HomeContainer.find("h1").text()).toBe("Testing");
  expect(HomeContainer).toMatchSnapshot();
});

test("About tests", () => {
  const AboutContainer = shallow(<About title="About Page" />);

  expect(AboutContainer.find("h1").text()).toBe("About Page");
  expect(AboutContainer).toMatchSnapshot();
});

test("App Interaction", () => {
  const AppContainer = mount(<App />);
  expect(AppContainer).toMatchSnapshot();

  AppContainer.find("a[href='/about']").first().simulate("click");
  expect(AppContainer.find("nav a.active").text()).toBe("About");

  AppContainer.find("a[href='/']").first().simulate("click");
  expect(AppContainer.find("nav a.active").text()).toBe("TinyProject");
});
