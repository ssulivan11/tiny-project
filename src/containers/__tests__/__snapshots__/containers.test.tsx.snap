// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About tests 1`] = `
Array [
  <h1>
    About Page
  </h1>,
  <p>
    This is just a simple
     
    <a
      href="https://github.com/preactjs/preact"
    >
      preact
    </a>
     typescript demo project messing with
     
    <a
      href="https://github.com/vuejs/vite#bare-module-resolving"
    >
      vite
    </a>
     for the dev server, stupid fast, no bundling and hmr.
  </p>,
  <p>
    Also utilizing
     
    <a
      href="https://github.com/parcel-bundler/parcel"
    >
      parcel
    </a>
     as the bundler. 
    <a
      href="https://github.com/facebook/jest"
    >
      Jest
    </a>
     and
     
    <a
      href="https://github.com/cypress-io/cypress"
    >
      cypress
    </a>
     for testing.
  </p>,
]
`;

exports[`App Interaction 1`] = `
<Component>
  <div>
    <nav
      className="nav"
    >
      <div
        className="container"
      >
        <K
          className="active pagename current"
          href="/"
          onClick={[Function]}
        >
          <a
            className="active pagename current"
            href="/"
            onClick={[Function]}
          >
            TinyProject
          </a>
        </K>
        <K
          className=""
          href="/about"
          onClick={[Function]}
        >
          <a
            className=""
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
        </K>
        <a
          href="https://github.com/ssulivan11/tiny-project"
        >
          Github
        </a>
      </div>
    </nav>
    <div
      className="container"
    >
      <undefined>
        <Component
          path="/"
          title="Welcome to Tiny Project!"
        >
          <div
            className="text-center"
          >
            <img
              alt="Smiley face"
              height="200"
              src="src/assets/android-chrome-512x512.png"
              style={
                Object {
                  "display": "block",
                  "margin": "0 auto",
                }
              }
              width="200"
            />
            <h1>
              Welcome to Tiny Project!
            </h1>
            <p>
              Read a little more 
              <K
                href="/about"
              >
                <a
                  href="/about"
                  onClick={[Function]}
                >
                  here
                </a>
              </K>
            </p>
          </div>
        </Component>
        <Component
          path="/about"
          title="About the Tiny Project"
        >
          <h1>
            About the Tiny Project
          </h1>
          <p>
            This is just a simple
             
            <a
              href="https://github.com/preactjs/preact"
            >
              preact
            </a>
             typescript demo project messing with
             
            <a
              href="https://github.com/vuejs/vite#bare-module-resolving"
            >
              vite
            </a>
             for the dev server, stupid fast, no bundling and hmr.
          </p>
          <p>
            Also utilizing
             
            <a
              href="https://github.com/parcel-bundler/parcel"
            >
              parcel
            </a>
             as the bundler. 
            <a
              href="https://github.com/facebook/jest"
            >
              Jest
            </a>
             and
             
            <a
              href="https://github.com/cypress-io/cypress"
            >
              cypress
            </a>
             for testing.
          </p>
        </Component>
      </undefined>
    </div>
  </div>
</Component>
`;

exports[`Home tests 1`] = `
<div
  className="text-center"
>
  <img
    alt="Smiley face"
    height="200"
    src="src/assets/android-chrome-512x512.png"
    style={
      Object {
        "display": "block",
        "margin": "0 auto",
      }
    }
    width="200"
  />
  <h1>
    Testing
  </h1>
  <p>
    Read a little more 
    <K
      href="/about"
    >
      here
    </K>
  </p>
</div>
`;
